### ISSUE TITLE
Trouble adding ADT data to Seurat object

### ISSUE THREAD

**Issue by baracus37** [satijalab/seurat#2540]
Hi,

I have clustered 13,057 cells using Seurat v3.1.2 and now I would like to add ADT data to this object. 
In order to do so, I have extracted the Cellnames of these 13,057 cells and used it as a whitelist for CITE-seq_Count. I imported the results from Cite-seq_count and tried adding them to the Seurat object 
```
>ini_vs_rem.ADT.data <- Read10X("/media/pgm/Data/tobias/243-NGS-030919-0/2_Cite-Seq_count/Cite-Seq_Count_ini_vs_rem/umi_count/",gene.column=1)
>ini_vs_rem[["ADT"]] <- CreateAssayObject(counts = ini_vs_rem.ADT.data)
Error: Cannot add a different number of cells than already present
> ncol(ini_vs_rem)
[1] 13507
> ncol(ini_vs_rem.ADT.data)
[1] 13473
```
 At this point I noticed, that Cite Seq Count apparently removed 34 cells from my dataset. At this point I wanted to remove these  34 cells from the Seurat object in order to match the number of cells. But as the cellnames differ between both, this did not work either
```
> ini_vs_rem <- subset(ini_vs_rem,cells = colnames(ini_vs_rem.ADT.data))
Error in CellsByIdentities(object = object, cells = cells) : 
  Cannot find cells provided
> head(colnames(ini_vs_rem))
[1] "AAACGAATCCATACTT-1" "AAACGCTAGGAAGTGA-1" "AAACGCTCAGCACGAA-1" "AAACGCTGTATCTCTT-1"
[5] "AAACGCTGTGCTGCAC-1" "AAACGCTTCGAAACAA-1"
> head(colnames(ini_vs_rem.ADT.data))
[1] "AGTCAACAGACTACGG" "AGACCATGTATTGACC" "TCTCCGACACCATAAC" "TTTGGTTGTGTGAATA"
[5] "CACTGTCGTCCTTAAG" "TATCCTACACCGGCTA"
```

So I wanted to remove the -1/-2 that cellranger adds to the Cellnames and Rename the Cells

```
>test <- substr(colnames(ini_vs_rem), 1, nchar(colnames(ini_vs_rem))-2)
>ini_vs_rem <- RenameCells(ini_vs_rem,new.names = test)
Error in `.rowNamesDF<-`(x, value = value) : invalid 'row.names' length
```

Unfortunately this failed as well. Removing the last two characters worked fine, but `RenameCells` failed.

Any help on how to get my ADT.data into my Seurat object would be greatly appreciated!

**Comment by andrewwbutler** [comment]
Hi,

Renaming the cells in `ini_v_rem` to remove the "-1/-2" seems like a reasonable approach (assuming that after you do that, there aren't any duplicate cell names). Could you post the `traceback` for that last error?

**Comment by baracus37** [comment]
Hi Andrew,

thanks a lot for your support! Below you can find the traceback of that last error. 

```
> ini_vs_rem <- RenameCells(ini_vs_rem,new.names = test)
Error in `.rowNamesDF<-`(x, value = value) : invalid 'row.names' length
> traceback()
13: stop("invalid 'row.names' length")
12: `.rowNamesDF<-`(x, value = value)
11: `row.names<-.data.frame`(`*tmp*`, value = value)
10: `row.names<-`(`*tmp*`, value = value)
9: `rownames<-`(x = `*tmp*`, value = new.names.vst)
8: FUN(X[[i]], ...)
7: lapply(X = Misc(object = object, slot = "vst.set"), FUN = function(x) {
       new.names.vst <- new.names[which(x = x$cells_step1 %in% Cells(x = object))]
       x$cells_step1 <- new.names.vst
       rownames(x = x$cell_attr) <- new.names.vst
       return(x)
   })
6: withCallingHandlers(expr, warning = function(w) invokeRestart("muffleWarning"))
5: suppressWarnings(Misc(object, slot = "vst.set") <- lapply(X = Misc(object = object, 
       slot = "vst.set"), FUN = function(x) {
       new.names.vst <- new.names[which(x = x$cells_step1 %in% Cells(x = object))]
       x$cells_step1 <- new.names.vst
       rownames(x = x$cell_attr) <- new.names.vst
       return(x)
   }))
4: RenameCells.Assay(object = object[[assay]], new.names = new.cell.names)
3: RenameCells(object = object[[assay]], new.names = new.cell.names)
2: RenameCells.Seurat(ini_vs_rem, new.names = test)
1: RenameCells(ini_vs_rem, new.names = test)
```

**Comment by baracus37** [comment]
I found the culprit. Apparently there were some cells that were present in the ADT dataset, but not in the Seurat object and vice versa.
So I had to run Cite-Seq-count on every sample individually, then add the cellranger suffix manually to the cellnames and merge both ADT matrices. I then had to find all cells that are present in both ADT and Seurat object via `intersect()` and then subset both to these cells. Then I was able to add the ADT data to the Seurat object.
